moquette-0.17/broker/src/main/java/io/moquette/broker/Server.java

```diff
@@ -181,52 +196,262 @@ public void startServer(IConfig config, List<? extends InterceptHandler> handler
         LOG.info("Moquette integration has been started successfully in {} ms", startTime);
 
         initialized = true;
     }
 
+    private static IQueueRepository initQueuesRepository(IConfig config, Path dataPath, H2Builder h2Builder) throws IOException {
+        final IQueueRepository queueRepository;
+        final String queueType = config.getProperty(IConfig.PERSISTENT_QUEUE_TYPE_PROPERTY_NAME);
+        if ("h2".equalsIgnoreCase(queueType)) {
+            LOG.info("Configuring H2 queue store");
+            queueRepository = h2Builder.queueRepository();
+        } else if ("segmented".equalsIgnoreCase(queueType)) {
+            LOG.info("Configuring segmented queue store to {}", dataPath);
+            final int pageSize = config.intProp(BrokerConstants.SEGMENTED_QUEUE_PAGE_SIZE, BrokerConstants.DEFAULT_SEGMENTED_QUEUE_PAGE_SIZE);
+            final int segmentSize = config.intProp(BrokerConstants.SEGMENTED_QUEUE_SEGMENT_SIZE, BrokerConstants.DEFAULT_SEGMENTED_QUEUE_SEGMENT_SIZE);
+            try {
+                queueRepository = new SegmentQueueRepository(dataPath, pageSize, segmentSize);
+            } catch (QueueException e) {
+                throw new IOException("Problem in configuring persistent queue on path " + dataPath, e);
+            }
+        } else {
+            final String errMsg = String.format("Invalid property for %s found [%s] while only h2 or segmented are admitted", IConfig.PERSISTENT_QUEUE_TYPE_PROPERTY_NAME, queueType);
+            throw new RuntimeException(errMsg);
+        }
+        return queueRepository;
+    }
+
+    private void collectAndSendTelemetryDataAsynch(IConfig config) {
+        final Thread telCollector = new Thread(() -> collectAndSendTelemetryData(config));
+        telCollector.start();
+    }
+
+    private void collectAndSendTelemetryData(IConfig config) {
+        final String uuid = checkOrCreateUUID(config);
+
+        final String telemetryDoc = collectTelemetryData(uuid);
+
+        try {
+            sendTelemetryData(telemetryDoc);
+        } catch (IOException e) {
+            LOG.info("Can't reach the telemetry collector");
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Original exception", e);
+            }
+        }
+    }
+
+    private String checkOrCreateUUID(IConfig config) {
+        final String storagePath = config.getProperty(IConfig.DATA_PATH_PROPERTY_NAME, "");
+        final Path uuidFilePath = Paths.get(storagePath, ".moquette_uuid");
+        if (Files.exists(uuidFilePath)) {
+            try {
+                return new String(Files.readAllBytes(uuidFilePath), StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                LOG.error("Problem accessing file path: {}", uuidFilePath, e);
+            }
+        }
+        final UUID uuid = UUID.randomUUID();
+        final FileWriter f;
+        try {
+            f = new FileWriter(uuidFilePath.toFile(), false);
+            f.write(uuid.toString());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This code contains a potential information leakage in the handling of the following *data in case of exception*: (UUID uuid) -&gt; (toString). 

**Fix**
Release data more selectively, or declassify the data before it&#39;s released. 

**More info**
[Learn more about information leakage](https://cwe.mitre.org/data/definitions/209.html) (external link).